### ISSUE TITLE
amplify config gets corrupt, cant run any cli commands

### ISSUE THREAD

**Issue by nirmay** [aws-amplify/amplify-cli#2360]
$ amplify add codegen --apiId myapiId       
✔ Getting API details
✖ Getting API details
Unexpected token o in JSON at position 1
$ amplify status

Current Environment: myenv

Unexpected token o in JSON at position 1
SyntaxError: Unexpected token o in JSON at position 1
    at JSON.parse (<anonymous>)
    at readJsonFile (/home/ubuntu/.nvm/versions/node/v10.16.2/lib/node_modules/@aws-amplify/cli/lib/extensions/amplify-helpers/read-json-file.js:11:17)
    at /home/ubuntu/.nvm/versions/node/v10.16.2/lib/node_modules/@aws-amplify/cli/lib/extensions/amplify-helpers/resource-status.js:279:35
    at step (/home/ubuntu/.nvm/versions/node/v10.16.2/lib/node_modules/@aws-amplify/cli/lib/extensions/amplify-helpers/resource-status.js:33:23)
    at Object.next (/home/ubuntu/.nvm/versions/node/v10.16.2/lib/node_modules/@aws-amplify/cli/lib/extensions/amplify-helpers/resource-status.js:14:53)
    at /home/ubuntu/.nvm/versions/node/v10.16.2/lib/node_modules/@aws-amplify/cli/lib/extensions/amplify-helpers/resource-status.js:8:71
    at new Promise (<anonymous>)
    at __awaiter (/home/ubuntu/.nvm/versions/node/v10.16.2/lib/node_modules/@aws-amplify/cli/lib/extensions/amplify-helpers/resource-status.js:4:12)
    at getResourceStatus (/home/ubuntu/.nvm/versions/node/v10.16.2/lib/node_modules/@aws-amplify/cli/lib/extensions/amplify-helpers/resource-status.js:273:12)
    at AmplifyToolkit.<anonymous> (/home/ubuntu/.nvm/versions/node/v10.16.2/lib/node_modules/@aws-amplify/cli/lib/extensions/amplify-helpers/resource-status.js:320:42)

Since then all amplify commands show the same error.
Before running codegen, I have auth configured and was able to push/status all fine. 

I am running this in a cloud9 environment and so reset the whole env and tried again, same issue. clearly, one of the required files have been corrupt.

my backend-config.json file looks like this-
$ cat backend-config.json 
{
    "auth": {
        "myauthname": {
            "service": "Cognito",
            "providerPlugin": "awscloudformation",
            "dependsOn": []
        }
    }
}
any other file I could check?
How do I reinstate my amplify env without losing what is already configured and how do I move forward?

**Comment by nirmay** [comment]
It might be that amplify-meta.json is corrupt.

$ cat \#current-cloud-backend/amplify-meta.json 
[object Object]$ cat backend/amplify-meta.json 
[object Object]$

**Comment by attilah** [comment]
@nirmay could you please tell us what version of the CLI are you using? Also can you provide repro steps for us?

The contents of the ```#current-cloud-backend``` directory contains the last pushed version of the files. Please check your S3 deployment bucket and tell me is that file has the same corrupted json content?

**Comment by kaustavghosh06** [comment]
@nirmay Did you try running ‘amplify env pull —restore’?

**Comment by garywu** [comment]
I can cause this consistently by issuing amplify add codegen --apiId xxxxxxx
After amplify init everything is normal both #current-cloud-backend and ampl,ify-meta.json looks normal. codegen changes both to [object Object].
this is the output
```
amplify add codegen --apiId xxxxxxxx
√ Getting API details
× Getting API details
Unexpected token o in JSON at position 1
```

**Comment by garywu** [comment]
Adding GraphQL API manually with just amplify add  then choose GraphQL works.

amplify --version
3.9.0

**Comment by nirmay** [comment]
@attilah @kaustavghosh06 
Thanks, I had to reset my cloud9 env to reproduce this.

```
$ amplify -v
3.9.0
```

Steps are:
```
npm install -g @aws-amplify/cli
amplify configure
amplify -v
amplify init
amplify status
amplify add auth
amplify push auth
```

we can safely ignore the auth add/push for this issue.

I downloaded the amplify pushed version from the S3 deployment bucket and the amplify-meta.json is safe or intact.
but, I cannot run `amplify env pull --restore` because the meta json is corrupt.

**Comment by pnadkar** [comment]
I got this working

Fire this command :` amplify add codegen -apiId <API ID>`
Note that I have used just one '-'

This will then throw the error like:
`Please download schema.graphql or schema.json and place in <your project path will be displayed here> before adding codegen when not in an amplify project`

I then went to my AWS AppSync and downloaded my schema.grpahql file and pasted it in the specified folder.

Last, fire the command `amplify add codegen -apiId <API ID>` again

It should work.

**Comment by attilah** [comment]
@pnadkar I'm sure you're not using the latest version of the CLI, because we're not accepting apiId with a single dash.

@nirmay the fix is on its way in #2381

**Comment by pnadkar** [comment]
> @pnadkar I'm sure you're not using the latest version of the CLI, because we're not accepting apiId with a single dash.
> 
> @nirmay the fix is on its way in #2381

Thanks. When would this be available for me to download and update?

**Comment by SebSchwartz** [comment]
Working great again, thanks.
